src/tests/fuzz/fuzz_decodebase64.c

```diff
@@ -0,0 +1,59 @@
+/**
+ * @file
+ * @author Shivani Bhardwaj <shivani@oisf.net>
+ * fuzz target for DecodeBase64
+ */
+
+#include "suricata-common.h"
+#include "suricata.h"
+#include "util-base64.h"
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size);
+
+static int initialized = 0;
+
+static void Base64FuzzTest(const uint8_t *src, size_t len, size_t dest_size)
+{
+    for (uint8_t mode = BASE64_MODE_RELAX; mode <= BASE64_MODE_RFC4648; mode++) {
+        uint8_t *dest = malloc(dest_size);
+        uint32_t consumed_bytes = 0;
+        uint32_t decoded_bytes = 0;
+
+        DecodeBase64(dest, dest_size, src, len, &consumed_bytes, &decoded_bytes, mode);
+    }
+}
+
+#define BIT_SHIFT_SIZE 24
+#define BLK_SIZE       4
+#define BYTE_SIZE      8
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+    if (initialized == 0) {
+        // Redirects logs to /dev/null
+        setenv("SC_LOG_OP_IFACE", "file", 0);
+        setenv("SC_LOG_FILE", "/dev/null", 0);
+        // global init
+        InitGlobal();
+        run_mode = RUNMODE_UNITTEST;
+        initialized = 1;
+    }
+
+    uint8_t shift = 0;
+    uint32_t dest_size = 0;
+    if (size > BLK_SIZE)
+        shift = BIT_SHIFT_SIZE;
```

Comment:
I do not understand this shift logic...

Reply:
See the following:
1. Read the number of bytes we receive from the fuzzer.
2. If the bytes received from fuzzer >= 4, take first 4 bytes, shift them one by one by 24, 16, 8, 0 to make a u32 size.
3. If the bytes received from fuzzer < 4, take whatever bytes there are and try to make a random size out of it.

Reply:
Looks too complex : I would expect a fixed number of bytes for the length

Reply:
ok. Keeping two bytes for length